integration-testing/onboard_test.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2024 Illumio, Inc. All Rights Reserved.
+
+package integrationtesting
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/spf13/viper"
+)
+
+// Cluster represents a Kubernetes cluster with relevant metadata.
+type Cluster struct {
+	ID            string `json:"id"`
+	IllumioRegion string `json:"illumio_region"`
+	Enabled       bool   `json:"enabled"`
+	Onboarded     bool   `json:"onboarded"`
+}
+
+// ClustersResponse represents the response containing cluster data.
+type ClustersResponse struct {
+	Clusters      []Cluster `json:"clusters"`
+	TotalSize     int       `json:"total_size"`
+	NextPageToken string    `json:"next_page_token"`
+	PrevPageToken string    `json:"prev_page_token"`
+	Page          int       `json:"page"`
+}
+
+type OffboardRequest struct {
+	ClusterIds []string `json:"cluster_ids"`
+}
+
+// Config holds configuration values needed for the API requests.
+type Config struct {
+	TenantId     string
+	CloudIdKey   string
+	CloudIdValue string
+	UserId       string
+}
+
+// NewHTTPClient returns a reusable HTTP client.
+func NewHTTPClient() *http.Client {
+	return &http.Client{}
+}
+
+// getAuthorizationHeader returns the Basic Authorization header using the provided credentials.
+func getAuthorizationHeader(key, secret string) string {
+	credentials := fmt.Sprintf("%s:%s", key, secret)
+	encodedCredentials := base64.StdEncoding.EncodeToString([]byte(credentials))
+	return fmt.Sprintf("Basic %s", encodedCredentials)
+}
+
+// buildRequest creates a new HTTP request with common headers.
+func buildRequest(method, url string, body io.Reader, config Config) (*http.Request, error) {
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Authorization", getAuthorizationHeader(config.CloudIdKey, config.CloudIdValue))
+	req.Header.Add("Accept", "*/*")
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("X-Tenant-Id", config.TenantId)
+	req.Header.Add("X-User-Id", config.UserId)
+
+	return req, nil
+}
+
+// fetchClusters makes an HTTP request to fetch a list of clusters.
+func fetchClusters(config Config) ([]Cluster, error) {
+	baseURL := "https://cloud.illum.io/api/v1/k8s_cluster" // Replace with actual URL
```

Comment:
Is this comment still relevant?
```suggestion
	baseURL := "https://cloud.illum.io/api/v1/k8s_cluster"
```